src/components/extensions/donate/DonateModal/donateForm/index.tsx

```diff
@@ -0,0 +1,171 @@
+import CommonExtensionModal from '@/components/extensions/common/CommonExtensionModal'
+import SelectInput from '@/components/inputs/SelectInput'
+import ProfilePreview from '@/components/ProfilePreview'
+import { getPostQuery } from '@/services/api/query'
+import { getAccountDataQuery } from '@/services/subsocial/evmAddresses'
+import { useExtensionModalState } from '@/stores/extension'
+import { useMessageData } from '@/stores/message'
+import { cx } from '@/utils/class-names'
+import BigNumber from 'bignumber.js'
+import { useEffect, useMemo, useState } from 'react'
+import { useDonateModalContext } from '../../DonateModalContext'
+import {
+  useBuildEvmBeforeSend,
+  useBuildSubtrateBeforeSend,
+} from '../../hooks/useBuildBeforeMessegeSend'
+import { DonateProps } from '../types'
+import { chainItems, tokensItems } from '../utils'
+import EvmDonateForm from './EvmDonateFormPart'
+import SubstrateDonateFormPart from './SubstrateDonateFormPart'
+
+const DonateForm = ({
+  chainState,
+  tokenState,
+  onSubmit,
+  hubId,
+  chatId,
+  setCurrentStep,
+  onSwitchButtonClick,
+}: DonateProps) => {
+  const { closeModal, isOpen, initialData } = useExtensionModalState(
+    'subsocial-donations'
+  )
+  const { showChatForm, disableButton } = useDonateModalContext()
+
+  const { messageId } = initialData
+  const setReplyTo = useMessageData((state) => state.setReplyTo)
+
+  useEffect(() => {
+    if (isOpen) setReplyTo(messageId)
+  }, [setReplyTo, messageId, isOpen])
+
+  const [selectedChain, setSelectedChain] = chainState
+  const [selectedToken, setSelectedToken] = tokenState
+
+  const [inputError, setInputError] = useState<string | undefined>()
+  const [amount, setAmount] = useState<string>('')
+
+  const amountPreview = amount
+    ? ` ${new BigNumber(amount).toFormat()} ${selectedToken.label}`
+    : ''
+
+  const evmBeforeMessageSend = useBuildEvmBeforeSend({
+    selectedChain,
+    selectedToken,
+    setCurrentStep,
+  })
+
+  const substrateBeforeMessageSend = useBuildSubtrateBeforeSend({
+    selectedChain,
+    selectedToken,
+    setCurrentStep,
+  })
```

Comment:
Why we are building both?

You can dot something like:

```
isEvmChain = chainKind === 'evm'
```

And use it in all places